modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/DataStreamLifecycleService.java

```diff
@@ -1126,7 +1127,19 @@ public void onResponse(Void unused) {
 
         @Override
         public void onFailure(Exception e) {
-            errorStore.recordError(targetIndex, e);
+            recordAndLogError(targetIndex, errorStore, e, errorLogMessage);
+        }
+    }
+
+    /**
+     * Records the provided error for the index in the error store and logs the error message at `ERROR` level if the error for the index
+     * is different to what's already in the error store.
+     * This allows us to not spam the logs and only log new errors when we're about to record them in the store.
+     */
+    static void recordAndLogError(String targetIndex, DataStreamLifecycleErrorStore errorStore, Exception e, String logMessage) {
+        String previousError = errorStore.recordError(targetIndex, e);
+        if (previousError == null || previousError.equals(errorStore.getError(targetIndex)) == false) {
```

Comment:
else should we log at trace level here? I'm just imagining a user / support person wanting a sanity check if they've seen no errors for a long time.

Reply:
Great idea.